packages/hash/frontend/src/blocks/page/BlockContextMenu/BlockContextMenuItem.tsx

```diff
@@ -0,0 +1,93 @@
+import { faChevronRight } from "@fortawesome/free-solid-svg-icons";
+import { ListItemIcon, ListItemText, MenuItem } from "@mui/material";
+import {
+  bindPopover,
+  bindHover,
+  bindFocus,
+  usePopupState,
+} from "material-ui-popup-state/hooks";
+import HoverPopover from "material-ui-popup-state/HoverPopover";
+import { cloneElement, forwardRef, RefObject, useRef, useState } from "react";
+import { FontAwesomeIcon } from "../../../shared/icons";
+
+type BlockContextMenuItemProps = {
+  itemKey: string;
+  onClick?: () => void;
+  icon: JSX.Element;
+  title: JSX.Element | string;
+  subMenu?: JSX.Element;
+  subMenuWidth?: number;
+};
+
+export const BlockContextMenuItem = forwardRef<
+  HTMLLIElement,
+  BlockContextMenuItemProps
+>(({ onClick, icon, title, itemKey, subMenu, subMenuWidth }, ref) => {
+  const subMenuPopupState = usePopupState({
+    variant: "popper",
+    popupId: `${itemKey}-submenu`,
+  });
+  const [subMenuOffsetTop, setSubmenuOffsetTop] = useState<
+    number | undefined
+  >();
+  const localRef = useRef<HTMLLIElement>(null);
+  const menuItemRef = (ref ?? localRef) as RefObject<HTMLLIElement>;
+
+  if (subMenu && !subMenuOffsetTop && menuItemRef.current) {
+    setSubmenuOffsetTop(menuItemRef.current.offsetTop);
+  }
+
+  return (
+    <MenuItem
+      ref={menuItemRef}
+      {...(subMenu
+        ? {
+            ...bindHover(subMenuPopupState),
+            ...bindFocus(subMenuPopupState),
+          }
+        : {
+            onClick,
+          })}
+      sx={{
+        position: "relative",
+      }}
+    >
+      <ListItemIcon>{icon}</ListItemIcon>
+      <ListItemText primary={title} />
+      {subMenu ? (
+        <>
+          <FontAwesomeIcon
+            icon={faChevronRight}
+            sx={({ palette }) => ({
+              ml: "auto",
+              color: palette.gray[50],
+              fontSize: 12,
+            })}
```

Comment:
Sure. Although we decided to stick to picking colors directly from our theme object because of the type safety we get and for consistency. It's outlined in this [internal document](https://www.notion.so/hashintel/Material-UI-Best-Practices-3031a2907df64274ac88aa57534d3e39)